includes/class-wc-payments-features.php

```diff
@@ -266,6 +266,12 @@ public static function is_payment_overview_widget_ui_enabled(): bool {
 	 * @return bool True if Direct Checkout is enabled, false otherwise.
 	 */
 	public static function is_woopay_direct_checkout_enabled() {
+		// If WooPayments is not enabled then disable Direct checkout.
+		$enabled_gateways = WC()->payment_gateways->get_available_payment_gateways();
+		if ( ! isset( $enabled_gateways['woocommerce_payments'] ) ) {
+			return false;
+		}
```

Comment:
@asumaran I wonder if we should move this check to `is_woopay_enabled`. This method also checks if `is_woopay_enabled`.

Reply:
We could then remove this manual check [here](https://github.com/Automattic/woocommerce-payments/blob/812ac5427676f4fcd06115d2c078405e001e075d/includes/class-wc-payments-woopay-button-handler.php#L249-L252) as well and leave only [is_woopay_enabled](https://github.com/Automattic/woocommerce-payments/blob/812ac5427676f4fcd06115d2c078405e001e075d/includes/class-wc-payments-woopay-button-handler.php#L254-L257).

Reply:
I was wondering wether to disable WooPay entirely or just the Direct Checkout feature and asked about it here p1721677050139319/1721674719.137809-slack-C01BZUL57SQ

But given that this method calls `is_woopay_enabled` I guess it just makes sense to move this check to that method instead. I'll update the code. FYI @lovo-h

Reply:
Updated the code here [370c16a](https://github.com/Automattic/woocommerce-payments/pull/9147/commits/370c16a586024cb840b65180fa685e8bb483155d) also added an additional test case.

Reply:
@ricardo do you mind taking a final look before merging it?